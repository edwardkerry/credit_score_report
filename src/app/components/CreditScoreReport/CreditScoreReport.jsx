import React, { Component } from 'react';
import { number, string } from 'prop-types';

import bemHelper from '../../utils/bem';

const cn = bemHelper({ block: 'creditScore' });

const CreditScoreReport = ({ score, maxScore, description }) => {
  return (
    <div className={cn('report', null)}>
      Your credit score is
        <p className={cn('score', null)}>{score}</p>
      out of
        <p className={cn('maxScore', null)}> {maxScore} </p>
      <p className={cn('description', null)}>{description}</p>
    </div>
  )
}

CreditScoreReport.propTypes = {
  description: string.isRequired,
  maxScore: number.isRequired,
  score: number.isRequired 
}

export default CreditScoreReport;