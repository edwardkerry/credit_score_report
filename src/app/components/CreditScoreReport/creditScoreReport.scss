.report__creditScore {
  .report__score { 
    color: #76e8a9;
    font-size: 50px;
    line-height: 0;

    @media only screen and (min-device-width: 320px) and (max-device-width: 480px) {
      font-size: 30px;
    }
  }
  
  .report__description {
    color: #76e8a9;
  }
}