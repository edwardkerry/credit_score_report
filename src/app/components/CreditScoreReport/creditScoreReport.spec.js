import React from 'react';
import renderer from 'react-test-renderer';
import CreditScoreReport from './CreditScoreReport.jsx';

describe('CreditScoreReport', () => {
  it('renders', () => {
    const score = 100;
    const maxScore = 700;
    const description = 'Not great'

    const component = renderer.create(
    <CreditScoreReport 
      score={score} 
      maxScore={maxScore} 
      description={description}/>
    );
    
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
