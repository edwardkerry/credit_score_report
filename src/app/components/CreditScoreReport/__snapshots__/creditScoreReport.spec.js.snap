// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreditScoreReport renders 1`] = `
<div
  className="creditScore__report"
>
  Your credit score is
  <p
    className="creditScore__score"
  >
    100
  </p>
  out of
  <p
    className="creditScore__maxScore"
  >
     
    700
     
  </p>
  <p
    className="creditScore__description"
  >
    Not great
  </p>
</div>
`;
